ckan/config/middleware/flask_app.py

```diff
@@ -65,6 +66,16 @@ def __call__(self, environ, start_response):
         return self.app(environ, start_response)
 
 
+class RepozeAdapterMiddleware(object):
+    def __init__(self, app):
+        self.app = app
+
+    def __call__(self, environ, start_response):
+        request = webob.Request(environ)
```

Comment:
cant we use the werkzeug Request  instead webob?

Reply:
+1, webob should be dropped soon

What does this middleware do anyway? AFAICT it just forwards the request

Reply:
Because we are using repoze.who which doesn't play well with werkzeug. And this adapter is required(previously Fanstatic had done it) in order to avoid BadRequests when our repoze.who interrupts unauthorized requests, dropping request payload while werkzeug tries to work with it.